### PR TITLE
tools: add the DM relay log doc

### PR DESCRIPTION
**Pull request by CaitinChen**
Via https://github.com/pingcap/tidb-tools/pull/153
@amyangfei @lilin90 PTAL

### CODE REVIEW

File: tools/dm-relay-log.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+    - DM-worker stores the binlogs synchronized from the upstream database in the same directory. Each directory is a `subdir`.
+
+    - `subdir` is named `<Upstream database UUID>.<Local subdir serial number>`.
+
+    - After [a switch between master and slave instances](../tools/data-migration-cluster-operations.md#switch-between-master-and-slave-instances) in the upstream, DM-worker generates a new `subdir` directory with an incremental serial number.
+    
+        - In the above example, for the `7e427cc0-091c-11e9-9e45-72b7c59d52d7.000001` directory, `7e427cc0-091c-11e9-9e45-72b7c59d52d7` is the upstream database UUID and `000001` is the local `subdir` serial number.
+
+- `server-uuid.index`: Records the list information of the current `subdir` directory.
```
Comment (amyangfei) [inline tools/dm-relay-log.md]: ``Records a list of names of currently available `subdir` directory.``

File: tools/dm-relay-log.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+## Initial synchronization rules
+
+For the start of DM-worker (or the relay log resuming synchronization after a pause), the starting position of synchronization is determined according to the concrete condition as follows:
+
+- If a valid local relay log (a valid relay log is a relay log with valid `server-uuid.index`, `subdir` and `relay.meta` files), DM-worker resumes synchronization from a position recorded by `relay.meta`.
+
+- If a valid local relay log does not exist, and `relay-binlog-name` or `relay-binlog-gtid` is not specified in the DM configuration file:
+
+    - In the non-GTID mode, DM-worker starts synchronization from the initial upstream binlog and synchronizes all the upstream binlog files to the latest local configuration directory successively.
```
Comment (amyangfei) [inline tools/dm-relay-log.md]:
```suggestion
    - In the non-GTID mode, DM-worker starts synchronization from the initial upstream binlog and synchronizes all the upstream binlog files to the latest successively.
```

**Comment by CaitinChen** [comment]
@amyangfei Thanks. PTAL again.

File: tools/dm-relay-log.md [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+
+- `purge-expires`
+
+    - The number of hours that a non-updated relay log can be retained for before being purged in the automatic background purge.
+    - "0" by default, indicating data purge is not performed according to the update time of the relay log.
+
+- `purge-remain-space`
+
+    - The amount of remaining disk space in GB less than which the specified DM-worker machine tries to purge the relay log that can be purged securely in the automatic background purge. If it is set to `0`, data purge is not performed according to the remaining disk space.
+    - `15` by default, indicating when the available disk space is less than 15GB, DM-master tries to purge the relay log securely.
```
Comment (lilin90) [inline tools/dm-relay-log.md]:
```suggestion
    - "15" by default, indicating when the available disk space is less than 15GB, DM-master tries to purge the relay log securely.
```

**Comment by CaitinChen** [comment]
@lilin90 PTAL again

**Pull request merged**